File: Csharp/AddressDirector.cs
namespace Builder_C_Sharp;

public class AddressDirector
{
    private AddressBuilder _addressBuilder;

    public AddressDirector(AddressBuilder addressBuilder) {
        _addressBuilder = addressBuilder;
    }

    public void BuildAddress(){
        _addressBuilder.BuildLine1();
        _addressBuilder.BuildLine2();
        _addressBuilder.BuildLine3();
        _addressBuilder.BuildLine4();
        _addressBuilder.BuildLine5();
        _addressBuilder.BuildLine6();
    }
}

File: TypeScript/AddressDirector.ts
import { AddressBuilder } from './AddressBuilder';

export class AddressDirector {

  constructor(private readonly addressBuilder : AddressBuilder) {}

  buildAddress() : void {
    this.addressBuilder.buildLine1();
    this.addressBuilder.buildLine2();
    this.addressBuilder.buildLine3();
    this.addressBuilder.buildLine4();
    this.addressBuilder.buildLine5();
    this.addressBuilder.buildLine6();
    this.addressBuilder.buildLine7();
    this.addressBuilder.buildLine8();
  }
}